File: JS/LeetC/strings/isAnagram.js
function isAnagram(str1, str2) {

	let [smaller, larger] = str1.length < str2.length ? [str1, str2] : [str2, str1];

	let obj = {};
	for (let c of smaller) {
		obj[c] = obj[c] ? obj[c] + 1 : 1;
	}

	for (let c of larger) {
		if (obj[c]) {
			obj[c] = obj[c] - 1;
		} else {
			return false;
		}
	}

	return true;
}

File: JS/LeetC/g75/isAnagram.py
class Solution:

    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        obj = dict()

        for c in s:
            obj[c] = obj.get(c, 0) + 1

        for c in t:
            if not obj.get(c):
                return False
            else:
                obj[c] -= 1

        return True
